views/Authentication/CreatePassword/CreatePassword.tsx

```diff
@@ -23,13 +23,18 @@ const CreatePassword: FC<ICreatePassword> = ({
   createPasswordNotification,
   addAuthNotification,
 }) => {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
   const { handleMsgType } = useNotification();
 
   const history = useHistory();
   const location = useLocation();
   const parsedRecoveryTokenData = parseRecoveryToken(location.search);
 
+  useEffect(() => {
+    if (parsedRecoveryTokenData?.language)
+      i18n.changeLanguage(parsedRecoveryTokenData?.language);
+  }, [i18n,parsedRecoveryTokenData?.language]);
```

Comment:
It should work fine by putting `[parsedRecoveryTokenData]` as dependency array. Also update condition:
```
if (parsedRecoveryTokenData?.language) {
      i18n.changeLanguage(parsedRecoveryTokenData.language);
}
```